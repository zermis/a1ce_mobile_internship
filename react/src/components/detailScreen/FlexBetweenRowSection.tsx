import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { detailPageStyle } from '../../shared/Styles';

interface Props {
    header?: string;
    body?: JSX.Element;
    style?: object;
}

const FlexBetweenRowSection = (props: Props): JSX.Element => {
    return (
        <View style={styles.flexBetweenContainer}>
            <Text style={{ ...detailPageStyle.textHeader, ...props.style }}>
                {props.header}
            </Text>
            {props.body}
        </View>
    );
};

const styles = StyleSheet.create({
    flexBetweenContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginVertical: 9
    }
});

export default FlexBetweenRowSection;
