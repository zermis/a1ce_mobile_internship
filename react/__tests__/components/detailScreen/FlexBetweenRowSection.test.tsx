import FlexBetweenRowSection from '../../../src/components/detailScreen/FlexBetweenRowSection';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

it('render flex between row section correctly', function () {
    var tree = renderer
        .create(
            <FlexBetweenRowSection header="Header" body={<Text>Body</Text>} />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
