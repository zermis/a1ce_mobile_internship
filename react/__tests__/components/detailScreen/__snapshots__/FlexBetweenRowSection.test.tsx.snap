// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render flex between row section correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 9,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#353334",
        "fontFamily": "ibm-semibold",
        "fontSize": 16,
        "marginBottom": 9,
      }
    }
  >
    Header
  </Text>
  <Text>
    Body
  </Text>
</View>
`;
